### PR TITLE
finish book model with listing all book

### PR DESCRIPTION
**Pull request by nguyenngodinh**

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    respond_to do |format|
+      format.html {redirect_to books_url, notice: 'Book was successfully destroyed.'}
+      format.json {head :no_content}
+    end
+  end
+
+  private
+  
+  def set_book
+    @book = Book.find(params[:id])
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: remove ()

File: app/views/books/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        <td><%= book.isbn %></td>
+        <td><%= book.title %></td>
+        <td><%= book.edition %></td>
+        <td><%= book.publish_date %></td>
+        <td><%= book.author %></td>
+        <td><%= book.number_of_pages %></td>
+        <td><%= link_to 'Show', book, class: 'btn btn-default' %></td>
+        <td><%= link_to 'Edit', edit_book_path(book), class: 'btn btn-default' %></td>
+        <td><%= link_to 'Destroy', book, class: 'btn btn-default',
+                method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (dieunb) [inline app/views/books/index.html.erb]:
```
 {confirm: 'Are you sure?'}
```

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,7 @@
 *.rbc
 *.sassc
+*.coffee
+books.scss
```
Comment (dungdm93) [inline .gitignore]: Remove ```*.scss``` and ```*.coffee``` file after rails generate.
Don't use this. :(

File: .gitignore [hunk 4]
```diff
@@ -27,7 +29,7 @@ config/initializers/secret_token.rb
 /vendor/bundle
 
 # these should all be checked in to normalise the environment:
-# Gemfile.lock, .ruby-version, .ruby-gemset
+Gemfile.lock, .ruby-version, .ruby-gemset
```
Comment (dungdm93) [inline .gitignore]: Dont't _ignore_ those file because I't need to deploy in Heroku.

File: app/assets/javascripts/books.coffee [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (dungdm93) [inline app/assets/javascripts/books.coffee]: Remove those file after ```rails generate```

File: app/views/books/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<%= form_for @book  do |f| %>
+  <% if @book.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@book.errors.count, "error") %> prohibited this book from being saved:</h2>
```
Comment (dungdm93) [inline app/views/books/_form.html.erb]: remove parens ```()```

File: app/views/books/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      <th>Edition</th>
+      <th>Publish date</th>
+      <th>Author</th>
+      <th>Number of pages</th>
+      <th colspan="3">Book operators</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @books.each do |book| %>
```
Comment (dungdm93) [inline app/views/books/index.html.erb]: I recomment render using partial
```_book.html.erb```

File: config/routes.rb [hunk 8]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  resources :books
```
Comment (dungdm93) [inline config/routes.rb]: move this line after this line ```devise_for :users```

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -18,7 +18,7 @@
       <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
         <ul class="nav navbar-nav">
           <li class="active">
-            <%= link_to '#' do %>
+            <%= link_to books_path do %>
```
Comment (dungdm93) [inline app/views/layouts/_header.html.erb]: ```Dashboard``` is for ```Activity```  not for ```Books```

File: .gitignore (hunk 3 above)
Comment (nguyenngodinh) [inline .gitignore]: ok. agree.

File: .gitignore (hunk 4 above)
Comment (nguyenngodinh) [inline .gitignore]: ok. as you want

File: app/assets/javascripts/books.coffee (hunk 5 above)
Comment (nguyenngodinh) [inline app/assets/javascripts/books.coffee]: ok. agree

File: app/views/books/_form.html.erb (hunk 6 above)
Comment (nguyenngodinh) [inline app/views/books/_form.html.erb]: ok. agree

File: app/views/books/index.html.erb (hunk 7 above)
Comment (nguyenngodinh) [inline app/views/books/index.html.erb]: I just want draw it as table.

File: app/assets/stylesheets/scaffolds.scss [hunk 10]
```diff
@@ -0,0 +1,69 @@
+body {
```
Comment (dungdm93) [inline app/assets/stylesheets/scaffolds.scss]: I recomment using original ```Bootstrap``` and don't override too much.
It can lead to conflic CSS.

File: app/views/books/index.html.erb (hunk 7 above)
Comment (dungdm93) [inline app/views/books/index.html.erb]: Create ```_ book.html.erb``
Then ```render @books```

File: config/routes.rb (hunk 8 above)
Comment (nguyenngodinh) [inline config/routes.rb]: in my branch, have not yet your routes devise_for :users. You can edit this after your branch be merged

File: db/migrate/20150326061907_create_books.rb [hunk 11]
```diff
@@ -0,0 +1,14 @@
+class CreateBooks < ActiveRecord::Migration
+  def change
+    create_table :books do |t|
+      t.integer :isbn
+      t.string :title
+      t.integer :edition
+      t.date :publish_date
+      t.string :author
```
Comment (dungdm93) [inline db/migrate/20150326061907_create_books.rb]: ```t.text :authors, array: true, default: []```
It's allows u store multiple authors. :+1:

File: app/views/books/index.html.erb (hunk 7 above)
Comment (nguyenngodinh) [inline app/views/books/index.html.erb]: ok. agree

File: config/routes.rb (hunk 8 above)
Comment (dungdm93) [inline config/routes.rb]: OK. :+1: 
I forgot it.

File: app/assets/stylesheets/scaffolds.scss (hunk 10 above)
Comment (nguyenngodinh) [inline app/assets/stylesheets/scaffolds.scss]: ok. i don't have any recognition about it. thanks

File: db/migrate/20150326061907_create_books.rb (hunk 11 above)
Comment (nguyenngodinh) [inline db/migrate/20150326061907_create_books.rb]: in design, we don't have any record database for author.
At the moment, I think it just need to save the author as string as enough.

File: app/views/layouts/_header.html.erb (hunk 9 above)
Comment (nguyenngodinh) [inline app/views/layouts/_header.html.erb]: how about home page for what. I think home page is most appropriate for activity. Because of our project name as book review system. show when the user first access to our app then they should know the review activity at first

File: db/migrate/20150326061907_create_books.rb (hunk 11 above)
Comment (dungdm93) [inline db/migrate/20150326061907_create_books.rb]: No extra table ```author```, and it's NOT reference (foreign key)
It's allows u store multiple authors in a column, just like ```serialize``` I told u.

Comment (nguyenngodinh) [inline db/migrate/20150326061907_create_books.rb]: I understand your idea, but do you think string is long enough to store the author names (may be a book has 5 authors at most)
http://stackoverflow.com/questions/3354330/difference-between-string-and-text-in-rails

File: .gitignore [hunk 12]
```diff
@@ -27,8 +27,8 @@ config/initializers/secret_token.rb
 /vendor/bundle
 
 # these should all be checked in to normalise the environment:
-# Gemfile.lock, .ruby-version, .ruby-gemset
-
+#Gemfile.lock, .ruby-version, .ruby-gemset
+Gemfile.lock
```
Comment (dungdm93) [inline .gitignore]: U can use below command to undo this change:
```git checkout ce7444794204520a0b64e92f13f7b5a1c4a83302 .gitignore```

File: app/views/books/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  </thead>
+
+  <tbody>
+    <%= render @books %>
+  </tbody>
+</table>
+
+<br>
+
+<%= link_to 'New Book', new_book_path, class: 'btn btn-default' %>
```
Comment (dungdm93) [inline app/views/books/index.html.erb]: Insert a new line at the end of file. Plz

File: app/views/books/_book.html.erb [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+  <td><%= book.title %></td>
+  <td><%= book.edition %></td>
+  <td><%= book.publish_date %></td>
+  <td><%= book.author %></td>
+  <td><%= book.number_of_pages %></td>
+  <td><%= link_to 'Show', book, class: 'btn btn-default' %></td>
+  <td><%= link_to 'Edit', edit_book_path(book), class: 'btn btn-default' %></td>
+  <td><%= link_to 'Destroy', book, class: 'btn btn-default',
+          method: :delete, data: {confirm: 'Are you sure?'} %></td>
+</tr>
```
Comment (dungdm93) [inline app/views/books/_book.html.erb]: Insert a new line at the end of file. Plz

File: db/migrate/20150326061907_create_books.rb (hunk 11 above)
Comment (dungdm93) [inline db/migrate/20150326061907_create_books.rb]: OK. ```string``` is fine :+1:

File: .gitignore (hunk 12 above)
Comment (nguyenngodinh) [inline .gitignore]: I :+1:

File: app/views/books/_book.html.erb (hunk 14 above)
Comment (nguyenngodinh) [inline app/views/books/_book.html.erb]: Can you give me some reasons for xyz,
Thanks.

Comment (dungdm93) [inline app/views/books/_book.html.erb]: Coding convention in [Framgia reference](https://github.com/styleguide/ruby)

Comment (nguyenngodinh) [inline app/views/books/_book.html.erb]: Ok. helpful
